src/config.cpp

```diff
@@ -81,6 +82,7 @@ void parse_program_options(int argc, const char** argv)
     bool kernel, no_kernel;
     bool list_clockids, list_events;
     std::uint64_t read_interval_ms;
+    boost::optional<std::uint64_t> metric_count, metric_frequency;
```

Comment:
So apparently, reading into `boost::optional<T>` is only supported since boost 1.65. I will think of another way to parse these mutually exclusive flags and at the same time give them sane defaults.

Reply:
Usually I'm the first one to update the dependency version, but there may be some concern with availability on wide spread distributions. boost 1.65 appears to be 3 months old. That said, Ubuntu LTS only comes with the ancient boost 1.58 which we don't currently support anyway... Ubuntu 17.* has 1.62.0. @bmario, @rschoene what do you think about bumping the boost version requirements?

Reply:
I'm against bumping boost version for such a minor issue. Just properly handle it in `lo2s::config::parse_program_options()`. There is `vm.count()`.

Reply:
I too think that this issue should *not* warrant updating boost.

The only caveat of `vm.count()` is that if `po::value::default_value()` is used on an option, `vm.count()` is always non-zero, even if that option was not specified. One could argue that this is the intended behavior, but right now I don't see how we would display a default value to the user that we ignore if the option was not specified.

<sup>But using `boost::optional` doesn't solve that issue either, I just realised</sup>